script.js

```diff
@@ -1,3 +1,49 @@
+/**
+ * AJAX request for users and groups
+ * Adapted from Struct plugin
+ *
+ * @param {function} fn Callback on success
+ * @param {string} id Call identifier
+ * @param {string} param Pass the parameter to backend
+ */
+function ajax_watchcycle(fn, id, param) {
+    let data = {};
+
+    data['call'] = 'plugin_watchcycle_' + id;
+    data['param'] = param;
+
+    jQuery.post(DOKU_BASE + 'lib/exe/ajax.php', data, fn, 'json')
+        .fail(function (result) {
+            if (result.responseJSON) {
+                if (result.responseJSON.stacktrace) {
+                    console.error(result.responseJSON.error + "\n" + result.responseJSON.stacktrace);
+                }
+                alert(result.responseJSON.error);
+            } else {
+                // some fatal error occurred, get a text only version of the response
+                alert(jQuery(result.responseText).text());
+            }
+        });
```

Comment:
Can this happen? If so, how would the end user deal with it? Is an alert really necessary here?